Enhance the e2e cleanup routines to check for volume pod deletion (Resiliency test suite)

dargasudarshan opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Enhance the e2e cleanup routines to check for volume pod deletion (Resiliency test suite).
- The current cleanup routines in e2e ansible playbooks delete the application specs (some check that the pvc are deleted). We need to mandate the check for pvc and corresponding pod deletions (there are possibilities that the pods might continue to exist even after pvc delete)
- This enhancement checks for successful deletion of volume pods.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1417 

**Special notes for your reviewer**:
@ksatchit Can you please review this PR?

ksatchit commented:
Can we do these steps for the replica failure tests also (this will cover the current iteration of changes into all available resiliency tests)

dargasudarshan commented:
Test playbook output post changes:

```
devops@devops-virtual-machine:~/openebs/e2e/ansible$ ansible-playbook pre-requisites.yml -vv
ansible-playbook 2.4.3.0
  config file = /home/devops/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/devops/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/devops/openebs/e2e/ansible/ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAYBOOK: pre-requisites.yml *********************************************************************************************************
1 plays in pre-requisites.yml

PLAY [localhost] *********************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************
ok: [localhost]
META: ran handlers

TASK [prerequisites : Install PIP Packages] ******************************************************************************************
task path: /home/devops/openebs/e2e/ansible/roles/prerequisites/tasks/main.yml:2
ok: [localhost -> 127.0.0.1] => (item=paramiko) => {"changed": false, "cmd": "/usr/local/bin/pip2 install paramiko", "item": "paramiko", "name": ["paramiko"], "requirements": null, "state": "present", "stderr": "", "stderr_lines": [], "stdout": "Requirement already satisfied: paramiko in /usr/lib/python2.7/dist-packages\n", "stdout_lines": ["Requirement already satisfied: paramiko in /usr/lib/python2.7/dist-packages"], "version": null, "virtualenv": null}
ok: [localhost -> 127.0.0.1] => (item=configparser) => {"changed": false, "cmd": "/usr/local/bin/pip2 install configparser", "item": "configparser", "name": ["configparser"], "requirements": null, "state": "present", "stderr": "", "stderr_lines": [], "stdout": "Requirement already satisfied: configparser in /usr/local/lib/python2.7/dist-packages\n", "stdout_lines": ["Requirement already satisfied: configparser in /usr/local/lib/python2.7/dist-packages"], "version": null, "virtualenv": null}

TASK [inventory : Generate Inventory] ************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/roles/inventory/tasks/main.yml:2
changed: [localhost -> 127.0.0.1] => {"changed": true, "cmd": "source ~/.profile; python /home/devops/openebs/e2e/ansible/roles/inventory/files/generate_inventory.py /home/devops/openebs/e2e/ansible/inventory/machines.in --passwordless=True", "delta": "0:00:07.760094", "end": "2018-04-11 12:02:53.688173", "rc": 0, "start": "2018-04-11 12:02:45.928079", "stderr": "", "stderr_lines": [], "stdout": "Inventory config generated successfully", "stdout_lines": ["Inventory config generated successfully"]}
META: inventory successfully refreshed
META: ran handlers
META: ran handlers

PLAY RECAP ***************************************************************************************************************************
localhost                  : ok=3    changed=1    unreachable=0    failed=0

devops@devops-virtual-machine:~/openebs/e2e/ansible$ ansible-playbook playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml -vvansible-playbook 2.4.3.0
  config file = /home/devops/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/home/devops/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/devops/openebs/e2e/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: The use of 'include' for tasks has been deprecated. Use 'import_tasks' for static inclusions or
'include_tasks' for dynamic inclusions. This feature will be removed in a future release. Deprecation warnings can be disabled by
setting deprecation_warnings=False in ansible.cfg.
[DEPRECATION WARNING]: include is kept for backwards compatibility but usage is discouraged. The module documentation details page
may explain more about this rationale.. This feature will be removed in a future release. Deprecation warnings can be disabled by
setting deprecation_warnings=False in ansible.cfg.
statically imported: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure-cleanup.yml
 [WARNING]: Ignoring invalid attribute: retried


PLAYBOOK: test-ctrl-failure.yml ******************************************************************************************************
1 plays in playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml

PLAY [localhost] *********************************************************************************************************************

TASK [Gathering Facts] ***************************************************************************************************************
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] ***********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:18
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.004168", "end": "2018-04-11 06:33:17.980566", "rc": 0, "start": "2018-04-11 06:33:17.976398", "stderr": "", "stderr_lines": [], "stdout": "/home/ubuntu", "stdout_lines": ["/home/ubuntu"]}

TASK [Get percona spec and liveness scripts] *****************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:25
changed: [localhost -> None] => (item=https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/percona.yaml) => {"changed": true, "checksum_dest": null, "checksum_src": "66bda981b5e740c1715ec7aefc89000f0797c403", "dest": "/home/ubuntu/percona.yaml", "gid": 1000, "group": "ubuntu", "item": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/percona.yaml", "md5sum": "d7b60974e8970cc7ceb7025ac028290f", "mode": "0664", "msg": "OK (1593 bytes)", "owner": "ubuntu", "size": 1593, "src": "/tmp/tmpWcrsoQ", "state": "file", "status_code": 200, "uid": 1000, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/percona.yaml"}
changed: [localhost -> None] => (item=https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/sql-test.sh) => {"changed": true, "checksum_dest": null, "checksum_src": "1808253d88759a699ea9738916d1febe04547ca6", "dest": "/home/ubuntu/sql-test.sh", "gid": 1000, "group": "ubuntu", "item": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/sql-test.sh", "md5sum": "d235f8179f540ebae853378070247fd5", "mode": "0664", "msg": "OK (399 bytes)", "owner": "ubuntu", "size": 399, "src": "/tmp/tmpH8U0F7", "state": "file", "status_code": 200, "uid": 1000, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/sql-test.sh"}

TASK [Replace volume-claim name with test parameters] ********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:32
changed: [localhost -> None] => (item=[u'demo-vol1-claim', u'test-ctrl-failure']) => {"changed": true, "item": ["demo-vol1-claim", "test-ctrl-failure"], "msg": "2 replacements made"}
changed: [localhost -> None] => (item=[u'demo-vol1', u'test-ctrl']) => {"changed": true, "item": ["demo-vol1", "test-ctrl"], "msg": "2 replacements made"}

TASK [Copy the chaoskube specs to kubemaster] ****************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:42
changed: [localhost -> None] => (item=rbac.yaml) => {"changed": true, "checksum": "b98d6989223026321514c47d9de3e2edc7b05577", "dest": "/home/ubuntu/rbac.yaml", "gid": 1000, "group": "ubuntu", "item": "rbac.yaml", "md5sum": "3c336e68b9a94fe87c73473e54daa52f", "mode": "0664", "owner": "ubuntu", "size": 419, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1523428411.76-222318890611184/source", "state": "file", "uid": 1000}
changed: [localhost -> None] => (item=chaoskube.yaml) => {"changed": true, "checksum": "adacd5ce5e7745ca47893b884b5da9966c7626e5", "dest": "/home/ubuntu/chaoskube.yaml", "gid": 1000, "group": "ubuntu", "item": "chaoskube.yaml", "md5sum": "30dbb0045350add58ebfcdc0a92d1b83", "mode": "0664", "owner": "ubuntu", "size": 1586, "src": "/home/ubuntu/.ansible/tmp/ansible-tmp-1523428413.43-54718257399417/source", "state": "file", "uid": 1000}

TASK [Start the log aggregator to capture test pod logs] *****************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:49
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; nohup stern \"maya*|openebs*|pvc*|percona*|chaos*\" --since 1m > \"/home/ubuntu/setup/logs/controller_failure_test.log\" &", "delta": "0:00:00.006162", "end": "2018-04-11 06:33:34.387283", "rc": 0, "start": "2018-04-11 06:33:34.381121", "stderr": "/bin/bash: /home/ubuntu/setup/logs/controller_failure_test.log: No such file or directory", "stderr_lines": ["/bin/bash: /home/ubuntu/setup/logs/controller_failure_test.log: No such file or directory"], "stdout": "", "stdout_lines": []}

TASK [Check whether maya-apiserver pod is deployed] **********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:57
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods | grep maya-apiserver", "delta": "0:00:00.276427", "end": "2018-04-11 06:33:34.979104", "rc": 0, "start": "2018-04-11 06:33:34.702677", "stderr": "", "stderr_lines": [], "stdout": "maya-apiserver-69f9db69-b9qxk          1/1       Running   0          19h", "stdout_lines": ["maya-apiserver-69f9db69-b9qxk          1/1       Running   0          19h"]}

TASK [Setup the chaoskube infrastructure] ********************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:73
changed: [localhost -> None] => (item=rbac.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f rbac.yaml", "delta": "0:00:00.357348", "end": "2018-04-11 06:33:35.691916", "item": "rbac.yaml", "rc": 0, "start": "2018-04-11 06:33:35.334568", "stderr": "", "stderr_lines": [], "stdout": "clusterrole \"chaoskube\" created\nclusterrolebinding \"chaoskube\" created", "stdout_lines": ["clusterrole \"chaoskube\" created", "clusterrolebinding \"chaoskube\" created"]}
changed: [localhost -> None] => (item=chaoskube.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f chaoskube.yaml", "delta": "0:00:00.360548", "end": "2018-04-11 06:33:36.379500", "item": "chaoskube.yaml", "rc": 0, "start": "2018-04-11 06:33:36.018952", "stderr": "", "stderr_lines": [], "stdout": "deployment \"chaoskube\" created\nserviceaccount \"chaoskube\" created", "stdout_lines": ["deployment \"chaoskube\" created", "serviceaccount \"chaoskube\" created"]}

TASK [Confirm that the chaoskube deployment is running] ******************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:80
FAILED - RETRYING: Confirm that the chaoskube deployment is running (15 retries left).
changed: [localhost -> None] => {"attempts": 2, "changed": true, "cmd": "source ~/.profile; kubectl get pods --no-headers -l app=chaoskube", "delta": "0:00:00.319918", "end": "2018-04-11 06:35:38.923233", "rc": 0, "start": "2018-04-11 06:35:38.603315", "stderr": "", "stderr_lines": [], "stdout": "chaoskube-55fc8f5f6d-xvkdh   1/1       Running   0         2m", "stdout_lines": ["chaoskube-55fc8f5f6d-xvkdh   1/1       Running   0         2m"]}

TASK [Create a configmap with the liveness sql script] *******************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:90
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl create configmap sqltest --from-file=sql-test.sh", "delta": "0:00:00.227770", "end": "2018-04-11 06:35:39.476957", "failed_when_result": false, "rc": 0, "start": "2018-04-11 06:35:39.249187", "stderr": "", "stderr_lines": [], "stdout": "configmap \"sqltest\" created", "stdout_lines": ["configmap \"sqltest\" created"]}

TASK [Create percona deployment with OpenEBS storage] ********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:98
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f percona.yaml", "delta": "0:00:00.856405", "end": "2018-04-11 06:35:40.665336", "rc": 0, "start": "2018-04-11 06:35:39.808931", "stderr": "", "stderr_lines": [], "stdout": "deployment \"percona\" created\npersistentvolumeclaim \"test-ctrl-failure\" created\nservice \"percona-mysql\" created", "stdout_lines": ["deployment \"percona\" created", "persistentvolumeclaim \"test-ctrl-failure\" created", "service \"percona-mysql\" created"]}

TASK [Wait for 120s to ensure liveness check starts] *********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:104
ok: [localhost] => {"changed": false, "elapsed": 120, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [Confirm percona pod is running] ************************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:108
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods --no-headers -l name=percona", "delta": "0:00:00.349647", "end": "2018-04-11 06:37:43.055912", "rc": 0, "start": "2018-04-11 06:37:42.706265", "stderr": "", "stderr_lines": [], "stdout": "percona-d8cbc694f-79q6j   1/1       Running   0         2m", "stdout_lines": ["percona-d8cbc694f-79q6j   1/1       Running   0         2m"]}

TASK [Set chaoskube pod name to variable] ********************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:118
ok: [localhost] => {"ansible_facts": {"chaospod": "chaoskube-55fc8f5f6d-xvkdh"}, "changed": false}

TASK [Get the name of the volume ctrl deployment] ************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:122
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get deployments -l openebs/controller=jiva-controller --no-headers", "delta": "0:00:01.788980", "end": "2018-04-11 06:37:45.181196", "rc": 0, "start": "2018-04-11 06:37:43.392216", "stderr": "", "stderr_lines": [], "stdout": "pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl   1         1         1         1         2m", "stdout_lines": ["pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl   1         1         1         1         2m"]}

TASK [Set the ctrl deployment name to variable] **************************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:131
ok: [localhost] => {"ansible_facts": {"ctrl_deploy": "pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl"}, "changed": false}

TASK [Get the resourceVersion of the controller deployment] **************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:141
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get deploy pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl -o yaml | grep resourceVersion | awk '{print $2}' | sed 's|\"||g'", "delta": "0:00:00.240140", "end": "2018-04-11 06:37:45.745040", "rc": 0, "start": "2018-04-11 06:37:45.504900", "stderr": "", "stderr_lines": [], "stdout": "102316", "stdout_lines": ["102316"]}

TASK [Initiate periodic controller failures from chaoskube] **************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:151
fatal: [localhost -> None]: FAILED! => {"changed": true, "cmd": "source ~/.profile; kubectl exec chaoskube-55fc8f5f6d-xvkdh -- timeout -t 120 chaoskube --labels 'openebs/controller=jiva-controller' --no-dry-run --interval=20s --debug", "delta": "0:02:00.793766", "end": "2018-04-11 06:39:46.854961", "msg": "non-zero return code", "rc": 143, "start": "2018-04-11 06:37:46.061195", "stderr": "time=\"2018-04-11T06:37:46Z\" level=debug msg=\"reading config\" annotations= debug=true dryRun=false excludedDaysOfYear= excludedTimesOfDay= excludedWeekdays= interval=20s kubeconfig= labels=\"openebs/controller=jiva-controller\" master= namespaces= timezone=UTC\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"starting up\" dryRun=false interval=20s version=v0.8.0\ntime=\"2018-04-11T06:37:46Z\" level=debug msg=\"using cluster config\" kubeconfig= master=\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"connected to cluster\" master=\"https://10.96.0.1:443\" serverVersion=v1.8.8\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"setting pod filter\" annotations= labels=\"openebs/controller=jiva-controller\" namespaces=\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"setting quiet times\" daysOfYear=\"[]\" timesOfDay=\"[]\" weekdays=\"[]\"\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"setting timezone\" location=UTC name=UTC offset=0\ntime=\"2018-04-11T06:37:46Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:37:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-lgvnb namespace=default\ntime=\"2018-04-11T06:37:46Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:38:06Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:38:06Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-8qt99 namespace=default\ntime=\"2018-04-11T06:38:06Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:38:26Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:38:26Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-9zc4b namespace=default\ntime=\"2018-04-11T06:38:26Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:38:46Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:38:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-9bzpg namespace=default\ntime=\"2018-04-11T06:38:46Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:39:06Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:39:06Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-zzfz4 namespace=default\ntime=\"2018-04-11T06:39:06Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:39:26Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:39:26Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-mv8gq namespace=default\ntime=\"2018-04-11T06:39:26Z\" level=debug msg=sleeping duration=20s\ntime=\"2018-04-11T06:39:46Z\" level=debug msg=\"found candidates\" count=1\ntime=\"2018-04-11T06:39:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-8pwcd namespace=default\ntime=\"2018-04-11T06:39:46Z\" level=debug msg=sleeping duration=20s\ncommand terminated with exit code 143", "stderr_lines": ["time=\"2018-04-11T06:37:46Z\" level=debug msg=\"reading config\" annotations= debug=true dryRun=false excludedDaysOfYear= excludedTimesOfDay= excludedWeekdays= interval=20s kubeconfig= labels=\"openebs/controller=jiva-controller\" master= namespaces= timezone=UTC", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"starting up\" dryRun=false interval=20s version=v0.8.0", "time=\"2018-04-11T06:37:46Z\" level=debug msg=\"using cluster config\" kubeconfig= master=", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"connected to cluster\" master=\"https://10.96.0.1:443\" serverVersion=v1.8.8", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"setting pod filter\" annotations= labels=\"openebs/controller=jiva-controller\" namespaces=", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"setting quiet times\" daysOfYear=\"[]\" timesOfDay=\"[]\" weekdays=\"[]\"", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"setting timezone\" location=UTC name=UTC offset=0", "time=\"2018-04-11T06:37:46Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:37:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-lgvnb namespace=default", "time=\"2018-04-11T06:37:46Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:38:06Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:38:06Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-8qt99 namespace=default", "time=\"2018-04-11T06:38:06Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:38:26Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:38:26Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-9zc4b namespace=default", "time=\"2018-04-11T06:38:26Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:38:46Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:38:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-9bzpg namespace=default", "time=\"2018-04-11T06:38:46Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:39:06Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:39:06Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-zzfz4 namespace=default", "time=\"2018-04-11T06:39:06Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:39:26Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:39:26Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-mv8gq namespace=default", "time=\"2018-04-11T06:39:26Z\" level=debug msg=sleeping duration=20s", "time=\"2018-04-11T06:39:46Z\" level=debug msg=\"found candidates\" count=1", "time=\"2018-04-11T06:39:46Z\" level=info msg=\"terminating pod\" name=pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl-559fc8b4f4-8pwcd namespace=default", "time=\"2018-04-11T06:39:46Z\" level=debug msg=sleeping duration=20s", "command terminated with exit code 143"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [Get the resourceVersion of the controller deployment] **************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:164
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get deploy pvc-8d82e3ae-3d52-11e8-83b2-02b983f0a4db-ctrl -o yaml | grep resourceVersion | awk '{print $2}' | sed 's|\"||g'", "delta": "0:00:00.304379", "end": "2018-04-11 06:39:47.533882", "rc": 0, "start": "2018-04-11 06:39:47.229503", "stderr": "", "stderr_lines": [], "stdout": "102861", "stdout_lines": ["102861"]}

TASK [Compare resourceVersions of controller deployments] ****************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:174
ok: [localhost] => {
    "failed_when_result": false,
    "msg": "Verified controller pods were restarted by chaoskube"
}

TASK [Confirm percona application is still running] **********************************************************************************
task path: /home/devops/openebs/e2e/ansible/playbooks/resiliency/test-ctrl-failure/test-ctrl-failure.yml:185
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods --no-headers -l name=percona", "delta": "0:00:00.228371", "end": "2018-04-11 06:39:48.070457", "rc": 0, "start": "2018-04-11 06:39:47.842086", "stderr": "", "stderr_lines": [], "stdout": "percona-d8cbc694f-79q6j   1/1       Running   1         4m", "stdout_lines": ["percona-d8cbc694f-79q6j   1/1       Running   1         4m"]}


```
